test/gululog_idx_SUITE.erl

```diff
@@ -138,6 +139,37 @@ t_scan_file_to_locate(Config) when is_list(Config) ->
   ?assertEqual(false, gululog_idx:locate(Dir, Idx0, 4)),
   ok.
 
+t_truncate_after({init, Config}) ->
+  Config;
+t_truncate_after({'end', _Config}) ->
+  ok;
+t_truncate_after(Config) when is_list(Config) ->
+  Dir = ?config(dir),
+  T1 = gululog_topic:init(Dir, []),
+  T2 = gululog_topic:append(T1, <<"key">>, <<"value">>),
+  T3 = gululog_topic:append(T2, <<"key">>, <<"value">>),
+  T4 = gululog_topic:append(T3, <<"key">>, <<"value">>),
+  T5 = gululog_topic:append(T4, <<"key">>, <<"value">>),
+  T6 = gululog_topic:append(T5, <<"key">>, <<"value">>),
+  T7 = gululog_topic:force_switch(T6),
+  T8 = gululog_topic:append(T7, <<"key">>, <<"value">>),
+  T9 = gululog_topic:force_switch(T8),
+  T10 = gululog_topic:append(T9, <<"key">>, <<"value">>),
+  T11 = gululog_topic:append(T10, <<"key">>, <<"value">>),
+  T12 = gululog_topic:append(T11, <<"key">>, <<"value">>),
+  Idx = erlang:element(4, T12),
+  ?assertEqual({ok, []}, gululog_idx:truncate_after(Dir, Idx, 10)),
+  ?assertEqual({ok, []}, gululog_idx:truncate_after(Dir, Idx, 8)),
+  ?assertEqual({ok, [{gululog_name:mk_idx_name(Dir, 6), gululog_name:mk_seg_name(Dir, 6)}]},
+               gululog_idx:truncate_after(Dir, Idx, 7)),
+  ?assertEqual({ok, [{gululog_name:mk_idx_name(Dir, 6), gululog_name:mk_seg_name(Dir, 6)}]},
+               gululog_idx:truncate_after(Dir, Idx, 6)),
+  Expect = [{gululog_name:mk_idx_name(Dir, 0), gululog_name:mk_seg_name(Dir, 0)},
+            {gululog_name:mk_idx_name(Dir, 5), gululog_name:mk_seg_name(Dir, 5)},
+            {gululog_name:mk_idx_name(Dir, 6), gululog_name:mk_seg_name(Dir, 6)}],
+  {ok, TruncateResult} = gululog_idx:truncate_after(Dir, Idx, 3),
```

Comment:
test append after truncation, then try to do a full-scan sanity check for the truncated file (as unit test).
the current code brutally truncate the file without close/reopen the index and segment writer fds, this might lead to:
1. corrupted file when trying to append to the topic after truncation.
2. failure if the fd pointed file has been deleted.
a side note: it can be even worse if we forget about the details and try to catch the exception then open new fds --- leak.